README.md

```diff
@@ -16,10 +16,199 @@ Alternatives for PHP:
 - [spid-php](https://github.com/italia/spid-php) based on [SimpleSAMLphp](https://simplesamlphp.org/)
 - [spid-php2](https://github.com/simevo/spid-php2) based on [php-saml](https://github.com/onelogin/php-saml)
 
+Framework specific libraries and examples based on spid-php-lib:
+- [https://github.com/italia/spid-symfony-bundle](https://github.com/italia/spid-symfony-bundle)
+- [https://github.com/simevo/spid-symfony3-example](https://github.com/simevo/spid-symfony3-example)
+- [https://github.com/simevo/spid-wordpress](https://github.com/simevo/spid-wordpress)
+
 Alternatives for other languages:
 - [spid-perl](https://github.com/italia/spid-perl)
 - [spid-ruby](https://github.com/italia/spid-ruby)
 
+
+
+Table of Contents
+=================
+
+* [Repository layout](#repository-layout)
+* [Getting started](#getting-started)
+    * [Prerequisites](#prerequisites)
+    * [Configuring & Installing](#configuring-and-installing)
+    * [Usage](#usage)
+        * [Performing login](#performing-login)
+        * [Performing logout](#performing-logout)
+        * [Complete API](#complete-api)
+    * [Example](#example)
+        * [Demo application](#demo-application)
+* [Features](#features)
+    * [More features](#more-features)
+* [Troubleshooting](#troubleshooting)
+* [Testing](#testing)
+    * [Unit Tests](#unit-tests)
+    * [Linting](#linting)
+* [Contributing](#contributing)
+* [See also](#see-also)
+* [Authors](#authors)
+* [License](#license)
+
+
+## Repository layout
+
+* [bin/](bin/) auxiliary scripts
+* [example/](example/) contains a demo application
+* [src/](src/) contains the library implementation
+* [test/](test/) contains the unit tests
+
+## Getting Started
+
+Tested on: amd64 Debian 9.5 (stretch, current stable) with PHP 7.0.
+
+Supports PHP 7.0, 7.1 and 7.2.
+
+### Prerequisites
+
+```sh
+sudo apt install composer make openssl php-curl php-zip php-xml phpunit
+```
+
+### Configuring and Installing
+
+
+**NOTE**: during testing, please use the test Identity Provider [spid-testenv2](https://github.com/italia/spid-testenv2).
+
+
+1. Install with composer 
+    
+    ```composer require italia/spid-php-lib```
+
+2. Generate key and certificate files for your Service Provider (SP).
+
+    Example: 
+    ```openssl req -x509 -nodes -sha256 -days 365 -newkey rsa:2048 -subj "/C=IT/ST=Italy/L=Milan/O=myservice/CN=localhost" -keyout sp.key -out sp.crt & wait;\```
+
+3. Download the Identity Provider (IdP) metadata files and place them in a directory in your project, for example `idp_metadata`. 
+    A convenience tool is provided to download those of the production IdPs: [bin/download_idp_metadata.php](bin/download_idp_metadata.php), example usage:
+    ```sh
+    bin/download_idp_metadata.php ./example/idp_metadata
```

Comment:
se ottieni la libreria con `composer require italia/spid-php-lib`, la dir di esempio al momento è in `vendor/italia/spid-php-lib/example` (e [non dovrebbe esserci](https://github.com/italia/spid-php-lib/issues/15)) mentre i binari sono in `vendor/italia/spid-php-lib/bin`

il comando di esempio quindi potrebbe essere:
```
mkdir idp_metadata
vendor/italia/spid-php-lib/bin/download_idp_metadata.php ./idp_metadata
```